favorites.cs

```diff
@@ -0,0 +1,155 @@
+
+                Console.WriteLine($"The query returned {queryResultsFilter.Count()} entities.");
+                
+                return new OkObjectResult(queryResultsFilter);
+
+            }
+            catch (Exception e)
+            {
+                log.LogError(e, "Problem loading");
+                return new BadRequestObjectResult("There was an issue");
+            }
+        }
+
+        [FunctionName("GetFavorites1")]
+        public async Task<IActionResult> RunGetFavorites1(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            try
+            {
+                string query = req.Query["UserId"];
+
+                Pageable<TableEntity> queryResultsFilter = _favoriteTableClient.Query<TableEntity>(filter: $"UserId eq {query}");
+
+                Console.WriteLine($"The query returned {queryResultsFilter.Count()} entities.");
+                
+                return new OkObjectResult(queryResultsFilter);
+
+            }
+            catch (Exception e)
+            {
+                log.LogError(e, "Problem loading");
+                return new BadRequestObjectResult("There was an issue");
+            }
+        }
+
+        [FunctionName("GetFavoritedItems")]
+        public async Task<IActionResult> RunGetFavoritedItems(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            try
+            {
+                string itemId = req.Query["ItemId"];
+
+                Console.WriteLine($"ItemId {itemId}");
+
+                Pageable<TableEntity> queryResultsFilter = _favoriteTableClient.Query<TableEntity>(filter: $"ItemId eq {itemId}");
+
+                Console.WriteLine($"The query returned {queryResultsFilter.Count()} entities.");
+                return new OkObjectResult(queryResultsFilter);
+
+            }
+            catch (Exception e)
+            {
+                log.LogError(e, "Problem loading");
+                return new BadRequestObjectResult("There was an issue");
+            }
+        }
+
+        [FunctionName("MakeFavorite")]
+        public async Task<IActionResult> RunMakeFavorite(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            try
+            {
+
+                string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+                log.LogInformation("Request body is {requestBody}", requestBody);
+                FavoriteItems favorites = JsonConvert.DeserializeObject<FavoriteItems>(requestBody);
+
+                await _favoriteTableClient.AddEntityAsync(favorites);
+
+                return new OkObjectResult(favorites);
+
+            }
+            catch (Exception e)
+            {
+                log.LogError(e, "Error with creating item. Check that it does not exist already.");
```

Comment:
If you wanted to prevent the table from restricting the same item from being uploaded again, you could override the  id with a GUI before adding it to the table? 

```
            Guid obj = Guid.NewGuid();    
            Console.WriteLine("New Guid is " + obj.ToString());  
```

Its not wrong if you keep doing it this why but just a suggestion!

Reply:
I'm not super familiar with Guid objects, so I'll need an explanation on this :)